### PR TITLE
imake: use gcc's preprocessor for imake

### PR DESCRIPTION
**Pull request by veprbl**
imake's default configuration files are not compatible with clang's C preprocessor, what gives runtime errors such as:

```
mv -f Makefile Makefile.bak
imake -DUseInstalled -I/usr/local/Cellar/imake/1.0.7/lib/X11/config
In file included from Imakefile.c:14:
In file included from /usr/local/Cellar/imake/1.0.7/lib/X11/config/Imake.tmpl:316:
/usr/local/Cellar/imake/1.0.7/lib/X11/config/Imake.rules:1674:27: warning: empty character constant [-Winvalid-pp-token]
        for flag in ${MAKEFLAGS} ''; do \                               @@\
                                 ^
In file included from Imakefile.c:14:
/usr/local/Cellar/imake/1.0.7/lib/X11/config/Imake.tmpl:2144:10: fatal error: ' X11 .rules' file not found
#include ProjectRulesFile
         ^
/usr/local/Cellar/imake/1.0.7/lib/X11/config/Imake.tmpl:2142:35: note: expanded from macro 'ProjectRulesFile'
# define ProjectRulesFile       Concat3(<,TopLevelProject,.rules>)
                                        ^
/usr/local/Cellar/imake/1.0.7/lib/X11/config/Imake.rules:256:23: note: expanded from macro 'Concat3'
#define Concat3(a,b,c)a/**/b/**/c
                      ^
1 warning and 1 error generated.
imake: Exit code 1.
  Stop.
```

This PR implements a patching similar to the one in https://trac.macports.org/browser/trunk/dports/x11/imake/files/patch-imakemdep.h.diff?rev=127961

### CODE REVIEW

File: imake.rb [hunk 1]
```diff
@@ -1,13 +1,17 @@
-require "formula"
-
 class Imake < Formula
   homepage "http://xorg.freedesktop.org"
   url "http://xorg.freedesktop.org/releases/individual/util/imake-1.0.7.tar.bz2"
   sha1 "52e236776133f217d438622034b8603d201a6ec5"
 
   depends_on "pkg-config" => :build
+  depends_on "gcc"
```
Comment (mistydemeo) [inline imake.rb]: Is this needed at runtime, or just at buildtime?

Comment (veprbl) [inline imake.rb]: Runtime.

**Comment by DomT4** [comment]
@BrewTestBot test this please

File: imake.rb (hunk 1 above)
Comment (DomT4) [inline imake.rb]: @mistydemeo Do we need to tag this dep as `depends_on "gcc" => :run` or?

**Comment by DomT4** [comment]
Presume this is okay as is, so have merged it in https://github.com/Homebrew/homebrew-x11/commit/d952e4dd3bbb07e6b57d0f03702e454a89688dac. It passed the bot, twice, at least. Thanks for your contribution to Homebrew, we appreciate it! :smiley_cat:

**Comment by veprbl** [comment]
@DomT4 Thank you!

**Pull request closed without merging**